### PR TITLE
Create BoolParam parameter type

### PR DESCRIPTION
**Pull request by AVB25**
Create basic implementation of boolean parameters, for easy handling of boolean arguments by parent fragments and scanning over booleans (not implemented). 

UnitTests included.

In future, add scans over bool parameters and present the results in two separate graphs for `True` and `False` cases. Similarly, could implement Enum parameters.

### CODE REVIEW

File: ndscan/experiment/parameters.py [hunk 1]
```diff
@@ -374,3 +417,36 @@ def eval_default(self, get_dataset: GetDataset) -> str:
...
+class BoolParam:
+    HandleType = BoolParamHandle
+    StoreType = BoolParamStore
+    CompilerType = TBool
+
+    def __init__(self,
+                 fqn: str,
+                 description: str,
+                 default: bool,
+                 is_scannable: bool = False):
```
Comment (dnadlinger) [inline ndscan/experiment/parameters.py:431]: This should probably default to `True`, as the idea for `is_scannable` is to be able to disable scans for a few parameters where one wants to make absolutely sure they are not scanned by accident, or the rare cases where some code can't handle scans.

**Comment by dnadlinger** [comment]
Thanks a lot, looks good! Could you address the one comment, and then perhaps fold the formatting commit/the extra change into the original three commits to keep the history easy to access? (If you haven't worked with rebasing/… before, we can discuss on Thursday.)

**Comment by dnadlinger** [comment]
Thanks a lot, looks good! Could you address the one comment, and then perhaps fold the formatting commit/the extra change into the original three commits to keep the history easy to access? (If you haven't worked with rebasing/… before, we can discuss on Thursday.)

**Comment by dnadlinger** [comment]
Thanks a lot, looks good! Could you address the one comment, and then perhaps fold the formatting commit/the extra change into the original three commits to keep the history easy to access? (If you haven't worked with rebasing/… before, we can discuss on Thursday.)

**Comment by dnadlinger** [comment]
We should totally look into which parts of all that duplicated code we can remove now that the ARTIQ compiler is quite a bit more stable overall.

File: ndscan/experiment/parameters.py (hunk 1 above)
Comment (AVB25) [reply ndscan/experiment/parameters.py:431]: Sure! I had defaulted it to `False` as the param doesn't currently handle scans, but I'll change that.

**Pull request merged**